src/docker/shared.rs

```diff
@@ -648,6 +657,50 @@ pub fn path_hash(path: &Path) -> Result<String> {
 #[cfg(test)]
 mod tests {
     use super::*;
+    use crate::id;
+
+    #[test]
+    fn test_docker_user_id() {
+        let var = "CROSS_ROOTLESS_CONTAINER_ENGINE";
+        let old = env::var(var);
+        env::remove_var(var);
+
+        let rootful = format!("\"engine\" \"--user\" \"{}:{}\"", id::user(), id::group());
+        let rootless = "\"engine\"".to_string();
+
+        let test = |engine, expected| {
+            let mut cmd = Command::new("engine");
+            docker_user_id(&mut cmd, engine);
+            assert_eq!(expected, &format!("{cmd:?}"));
+        };
+        test(EngineType::Docker, &rootful);
+        test(EngineType::Podman, &rootless);
+        test(EngineType::PodmanRemote, &rootless);
+        test(EngineType::Other, &rootless);
+
+        env::set_var(var, "0");
+        test(EngineType::Docker, &rootful);
+        test(EngineType::Podman, &rootful);
+        test(EngineType::PodmanRemote, &rootful);
+        test(EngineType::Other, &rootful);
+
+        env::set_var(var, "1");
+        test(EngineType::Docker, &rootless);
+        test(EngineType::Podman, &rootless);
+        test(EngineType::PodmanRemote, &rootless);
+        test(EngineType::Other, &rootless);
+
+        env::set_var(var, "auto");
+        test(EngineType::Docker, &rootful);
+        test(EngineType::Podman, &rootless);
+        test(EngineType::PodmanRemote, &rootless);
+        test(EngineType::Other, &rootless);
+
+        match old {
+            Ok(v) => env::set_var(var, v),
+            Err(_) => env::remove_var(var),
+        }
```

Comment:
vars are not shared in test context

Reply:
Sorry!! I think this is false actually. Dismiss this :)

Reply:
Whoops, Ok I'll re-apply this.